pkg/config/ca/ca.go

```diff
@@ -0,0 +1,109 @@
+package ca
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/viper"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	"github.com/jaegertracing/jaeger-operator/pkg/util"
+)
+
+// Get returns a trusted CA bundle configmap if platform is OpenShift
+func Get(jaeger *v1.Jaeger) *corev1.ConfigMap {
+	// Only configure the trusted CA if running in OpenShift
+	if viper.GetString("platform") != v1.FlagPlatformOpenShift {
+		return nil
+	}
+
+	if !deployTrustedCA(jaeger) {
+		jaeger.Logger().Debug("CA: Skip deploying the Jaeger instance's trustedCABundle configmap")
+		return nil
+	}
+
+	jaeger.Logger().Debug("CA: Creating the trustedCABundle configmap")
+	trueVar := true
+
+	name := TrustedCAName(jaeger)
+	labels := util.Labels(name, "ca-configmap", *jaeger)
+	labels["config.openshift.io/inject-trusted-cabundle"] = "true"
+
+	// See https://docs.openshift.com/container-platform/4.4/networking/configuring-a-custom-pki.html#certificate-injection-using-operators_configuring-a-custom-pki
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: jaeger.Namespace,
+			Labels:    labels,
+			OwnerReferences: []metav1.OwnerReference{{
+				APIVersion: jaeger.APIVersion,
+				Kind:       jaeger.Kind,
+				Name:       jaeger.Name,
+				UID:        jaeger.UID,
+				Controller: &trueVar,
+			}},
+		},
+		Data: map[string]string{
+			"ca-bundle.crt": "",
```

Comment:
Tests are showing that this is fine, but I wonder if there's a reason to have this as empty first? Also, would the reconciliation phase revert this to empty? If so, OpenShift seems to be refilling this quite fast (possibly with a webhook, before changes are actually applied).

Reply:
might be an idea to check with `kubectl get -w configmap -o yaml` - could be a potencial race?